drivers/serial/uart_xmc4xxx.c

```diff
@@ -57,21 +66,197 @@ static int uart_xmc4xxx_init(const struct device *dev)
 	XMC_GPIO_SetMode(P0_1,
 			 XMC_GPIO_MODE_OUTPUT_PUSH_PULL | P0_1_AF_U1C1_DOUT0);
 
+#if defined(CONFIG_UART_INTERRUPT_DRIVEN)
+	config->irq_config_func(dev);
+#endif
+
 	return 0;
 }
 
+#if defined(CONFIG_UART_INTERRUPT_DRIVEN)
+
+static void uart_xmc4xxx_isr(void *arg)
+{
+	const struct device *dev = arg;
+	struct uart_xmc4xxx_data *data = dev->data;
+
+	if (data->user_cb) {
+		data->user_cb(dev, data->user_data);
+	}
+}
+
+static int uart_xmc4xxx_fifo_fill(const struct device *dev, const uint8_t *tx_data, int len)
+{
+	const struct uart_xmc4xxx_config *config = dev->config;
+
+	XMC_UART_CH_Transmit(config->uart, tx_data[0]);
```

Comment:
This looks like it only sends the first byte of `tx_data` of length `len`. You will need to loop through the whole buffer to send all data.

Reply:
good point, updated to loop over the data when we are using the fifo.

Reply:
Please pull the looping code from later commit into this. So that if someone does git bisect, they still have a working UART driver.